src/backends/plonky2/common.rs

```diff
@@ -0,0 +1,95 @@
+//! Common functionality to build Pod circuits with plonky2
+
+use crate::middleware::STATEMENT_ARG_F_LEN;
+use crate::middleware::{Params, Value};
+use plonky2::field::extension::Extendable;
+use plonky2::field::types::PrimeField64;
+use plonky2::hash::hash_types::RichField;
+use plonky2::iop::target::{BoolTarget, Target};
+use plonky2::plonk::circuit_builder::CircuitBuilder;
+
+#[derive(Copy, Clone)]
+pub struct ValueTarget {
+    pub elements: [Target; 4],
+}
+
+#[derive(Clone)]
+pub struct StatementTarget {
+    pub code: [Target; 6],
+    pub args: Vec<[Target; STATEMENT_ARG_F_LEN]>,
+}
+
+impl StatementTarget {
+    pub fn to_flattened(&self) -> Vec<Target> {
+        self.code
+            .iter()
+            .chain(self.args.iter().flatten())
+            .cloned()
+            .collect()
+    }
+}
+
+// TODO: Implement Operation::to_field to determine the size of each element
+#[derive(Clone)]
+pub struct OperationTarget {
+    pub code: [Target; 6],
```

Comment:
This `6` comes from `Params::predicate_size()`, which returns `HASH_SIZE + 2`. Maybe we could find a way to avoid having the hardcoded 6 in this file and use it from a basetypes constantant
(mentioning this not because it is an issue having the `6` hardcoded now, but thinking into having a single 'source of truth' of the value, so that if in the future somebody changes the 'operation code length', the other parts that depend on that definition already get the correct value).
Same for the StatementTarget's code.